buildSrc/src/main/groovy/org/elasticsearch/gradle/UpgradableVersion.groovy

```diff
@@ -0,0 +1,117 @@
+package org.elasticsearch.gradle
+
+/**
+ * A naive version parser that is used to know if a given version can be upgraded to another version in the packaging tests
+  */
+class UpgradableVersion {
+
+    final int major;
+    final int minor;
+    final int revision;
+    final Stage stage;
+    final int extra;
+
+    public UpgradableVersion(int major, int minor, int revision, Stage stage, int extra) {
+        this.major = major
+        this.minor = minor
+        this.revision = revision
+        this.stage = stage
+        this.extra = extra
+    }
+
+    public boolean canBeUpgradedTo(String version) {
+        return canBeUpgradedTo(fromVersion(version))
+    }
+
+    boolean isStaging() {
+        return stage != null;
+    }
+
+    public boolean canBeUpgradedTo(UpgradableVersion version) {
+        // We can't upgrade from version before 5.0.0-alpha4 because of https://github.com/elastic/elasticsearch/pull/18554
+        if (this.major == 5 && this.minor == 0 && this.revision == 0 && this.stage == Stage.ALPHA && this.extra <= 4) {
```

Comment:
Or, better yet, we can only upgrade from the package current package until we hit a GA release.